rhombus-lib/rhombus/private/amalgam/boolean-annotation.rkt

```diff
@@ -17,15 +17,17 @@
 ;; ----------------------------------------
 ;; &&
 
-(define (handle-and/c form)
+(define (handle-combine/c sep-op-sep form)
   (and (pair? (cdr form))
        (list? (cdr form))
        (or (get-primitive-subcontract (cdr form))
```

Comment:
This also needs to be factored out.

Reply:
Sorry, I should’ve made this clearer: `get-primitive-subcontract` needs to be factored out, because this function specifically is only intended for `and/c` contracts, for rewriting contracts like `(and/c hash? (not/c immutable?))` into `mutable-hash?` (which then gets further rewritten).  Maybe we can have something like `get-primitive-supercontract`, if that makes sense for `or/c` contracts, but I don’t see any `or/c` contract that needs that treatment yet.

Reply:
Ah, I had not noticed or understood the "subcontract" part. Meanwhile, I made `get-primitive-contract` check for a complete match before trying the combinator table. Does it make sense to use the contract table that way, instead of having a subcontract table? If necessary in the future, a combination-handling function can try building and looking up combinations of subsets of components to handle something like `(and/c X? hash? (not/c immutable?))`.

Reply:
I added a commit to show what I mean.